tracer/src/Datadog.Trace/Ci/IntelligentTestRunnerClient.cs

```diff
@@ -703,17 +665,18 @@ private async Task<ObjectPackFilesResult> GetObjectsPackFileFromWorkingDirectory
             // Error handling block
             if (exceptionDispatchInfo is not null)
             {
+                var sourceException = exceptionDispatchInfo.SourceException;
+
                 if (isFinalTry)
```

Comment:
I would probably check if this is a "cancel" `RateLimitException` here, so we write an error log?
```suggestion
                var sourceException = exceptionDispatchInfo.SourceException;

                if (isFinalTry || sourceException is RateLimitException { DelayTimeInSeconds: null })
```

Reply:
done in [b4dd46d](https://github.com/DataDog/dd-trace-dotnet/pull/3490/commits/b4dd46dabd3c99f8047cbb92342899da6df8f9e4)